enzy_htp/_interface/pymol_interface.py

```diff
@@ -519,6 +522,113 @@ def get_sequence(self, session: pymol2.PyMOL, fname: str, sele: str = '(all)') -
         lines = list(filter(lambda ll: ll[0] != '>', lines))
         return ''.join(lines)
 
+    def create_cluster(self, session, fname: str, sele_str: str, outfile: str = None, cap_strategy: str = 'H', work_dir: str = None) -> str:
+        """TODO(CJ)"""
+
+        if work_dir is None:
+            work_dir = eh_config['system.WORK_DIR']
+
+        fs.check_file_exists(fname)
+
+        if outfile is None:
+            temp_path = Path(fname)
+            outfile: str = f"{work_dir}/{temp_path.stem}_cluster{temp_path.suffix}"
+
+        #TODO(CJ): add file check for fname
+        obj_name: str = '__eh_cluster'
+        self.general_cmd(session, [('delete', 'all'), ('load', fname), ('select', sele_str), ('create', obj_name, sele_str),
+                                   ('delete', Path(fname).stem)])
+
+        df: pd.DataFrame = self.collect(session, 'memory', "chain resi resn name".split())
+
+        args = list()
+        for i, row in df.iterrows():
+            if row['name'] not in "N C".split():
+                continue
+
+            if not row.resn in THREE_LETTER_AA_MAPPER:
+                continue
+
+            args.extend([('valence', 'guess', f"chain {row.chain} and resi {row.resi} and resn {row.resn} and name {row['name']}"),
+                         ('h_add', f"chain {row.chain} and resi {row.resi} and resn {row.resn} and name {row['name']}")])
+
+        args.append(("save", outfile, obj_name))
+
+        self.general_cmd(session, args)
+
+        if cap_strategy == 'H':
+            return outfile
+
+        if cap_strategy == 'CH3':
+
+            args = []
+            orig = set(zip(df.chain, df.resi, df.resn, df['name']))
+            updated: pd.DataFrame = self.collect(session, 'memory', "chain resi resn name".split())
+            new = set(zip(updated.chain, updated.resi, updated.resn, updated['name']))
+
+            for (cname, res_num, res_name, aname) in filter(lambda x: x not in orig, new):
+                if aname == 'H01':
+                    new_name = 'C21'
+                elif aname == 'H02':
+                    new_name = 'C22'
+                else:
+                    #TODO(CJ): better error message
+                    self.general_cmd(session, [('save', '_____mess_up.pdb')])
+                    assert False, (cname, res_num, res_name, aname)
+                args.extend([
+                    ('alter', f"chain {cname} and resi {res_num} and resn {res_name} and name {aname}", "elem='C'"),
+                    ('alter', f"chain {cname} and resi {res_num} and resn {res_name} and name {aname}", f"name='{new_name}'"),
+                ])
+
+            self.general_cmd(session, args)
+
+            args = [('valence', 'guess', 'name C21 or name C22'), ('h_add', 'name C21'), ('h_add', 'name C22'), ("save", outfile, obj_name)]
+            self.general_cmd(session, args)
+
+        return outfile
+
+
+    def center_of_mass(self, session, sele:str='all', no_hydrogens:bool=True):
```

Comment:
This can be realized by Structure() inside EnzyHTP btw

Reply:
good to know! sorry for duplciation.